### PR TITLE
Add more flexibility to JTable ordering

### PR DESCRIPTION
**Pull request by davidjosephhayes**
Pull Request for Issue # .
### Summary of Changes

Set JTable and JModelAdmin to use the JTable::_columnAlias property for ordering. Defaults to original behavior when there is no ordering property of JTable::_columnAlias.

Basically this implements what the function documentation for JTable::getColumnAlias says:
     \* Method to return the real name of a "special" column such as ordering, hits, published
     \* etc etc. In this way you are free to follow your db naming convention and use the
     \* built in Joomla functions.
### Testing Instructions

Use any built in component and reorder items.
### Documentation Changes Required

None.

### CODE REVIEW

File: libraries/joomla/table/table.php [hunk 1]
```diff
@@ -1287,14 +1287,15 @@ public function isCheckedOut($with = 0, $against = null)
 	public function getNextOrder($where = '')
 	{
 		// If there is no ordering field set an error and return false.
-		if (!property_exists($this, 'ordering'))
+		$orderingField = $this->getColumnAlias('ordering');
```
Comment (brianteeman) [inline libraries/joomla/table/table.php]: Surely this is not doing what the doc block above it says?

Comment (davidjosephhayes) [inline libraries/joomla/table/table.php]: How would you change the doc block?

**Comment by davidjosephhayes** [comment]
I update the code comments where it is throwing an exception instead of setting an error and returning false.

**Comment by davidjosephhayes** [comment]
Is there something else I should change?

**Comment by davidjosephhayes** [comment]
bump

**Comment by andrepereiradasilva** [comment]
needs two successfull tests to get merged in the core.

**Comment by csthomas** [comment]
* IMHO you should not call `quoteName($orderingField)` more than once per one method. Create a variable and use it in other places.

* You missed one https://github.com/joomla/joomla-cms/pull/12464/files#diff-8254e2c441a41d3fa26f0f83fbbb3043R1477

**Comment by davidjosephhayes** [comment]
Both should be fixed now @csthomas

**Comment by RonakParmar** [comment]
I have tested this item :white_check_mark: successfully on fd08d7b636c59b97626cd177196171781385e1ab<br /><br />Tested ordering of articles before and after PR, in both case ordering working fine. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12464">issues.joomla.org/tracker/joomla-cms/12464</a>.</sub>

**Comment by csthomas** [comment]
What about other files:
```
libraries/joomla/table/nested.php:1299:                 if (property_exists($this, 'ordering'))
libraries/joomla/table/asset.php:168:                   if (property_exists($this, 'ordering'))
```

I'm not sure but IMHO nested.php should be chnaged too.

**Comment by davidjosephhayes** [comment]
Never used those classes. I bet you are right. 

I will run through the rest of the libraries and see if there are others.

**Comment by mbabker** [comment]
Only JTable and JTableNested need to have this abstraction layer.  The other subclasses are "final" implementations for each table.

**Comment by davidjosephhayes** [comment]
I added the abstraction layer to JTableNested.

I feel like it should be on JTableAsset as well @mbabker since it is using a variable for the table name.

**Comment by mbabker** [comment]
A lot of the JTable subclasses reference the class variables in the methods versus hardcoding things like the table key(s) or name.  JTableAsset is the class directly mapping to the `#__assets` table so it shouldn't need direct mapping like the other two classes have.  Unfortunately, we neither declare the base classes (JTable or JTableNested) as abstract nor the "final" subclasses as final, so by that argument every column name supported by the abstraction layer should technically be mapped in the same way in all subclasses in case they are subclassed further by other developers.  Personal opinion, I don't think it's needed for instances where it's supposed to be a final object, but I'll let someone else make the final judgement call on it.

**Comment by davidjosephhayes** [comment]
Gotcha. I end up extending core classes a lot, so its always nice when the abstraction is already there. However, abstracting everything is likely out of the scope of this PR.

File: libraries/joomla/table/nested.php [hunk 2]
```diff
@@ -1296,9 +1296,10 @@ public function rebuild($parentId = null, $leftId = 0, $level = 0, $path = '')
 				->where('parent_id = %d');
 
 			// If the table has an ordering field, use that for ordering.
-			if (property_exists($this, 'ordering'))
+			$orderingField = $this->getColumnAlias('ordering');
+			if (!property_exists($this, $orderingField))
```
Comment (csthomas) [inline libraries/joomla/table/nested.php]: You have to remove exclamation mark.

**Comment by csthomas** [comment]
I tested successful 
and on review I have not found any ordering column in JTableNested subclasses. 
It is not used by joomla at all but because of B/C it probably has to stay. 
That OK.

Code style:
I thought that before `if ()` code line there should be a comment or empty line.
Am I wrong?

**Comment by davidjosephhayes** [comment]
No idea. I don't see anything in here though however: http://joomla.github.io/coding-standards/?coding-standards/chapters/php.md

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on 32a6436454b6f5c82e62fa120a0306315f8c6104<br /><br />Joomla works normal.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12464">issues.joomla.org/tracker/joomla-cms/12464</a>.</sub>

**Comment by davidjosephhayes** [comment]
How does the HumanTestResults test work?

**Comment by dgt41** [comment]
I have tested this item :white_check_mark: successfully on 32a6436454b6f5c82e62fa120a0306315f8c6104<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12464">issues.joomla.org/tracker/joomla-cms/12464</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12464">issues.joomla.org/tracker/joomla-cms/12464</a>.</sub>

**Comment by zero-24** [comment]
@davidjosephhayes you can take a look here: https://docs.joomla.org/Testing_Joomla!_patches#Recording_test_results

It is a feature of the issue tracker arround GitHub :)

**Comment by zero-24** [comment]
With the last 4 commit i have fixed some CS Issue travis did not mention (why?) and updated this branch to the last staging.

**Comment by zero-24** [comment]
@rdeutz @wilsonge please take a final review and merge decision here.

**Comment by jeckodevelopment** [comment]
@davidjosephhayes could you please look at the conflicts?

**Comment by davidjosephhayes** [comment]
Is it just me or is the conflict page not loading?

**Comment by csthomas** [comment]
For me it is not working too.

**Comment by jeckodevelopment** [comment]
@davidjosephhayes this is the conflicting file:
` libraries/joomla/table/table.php `

**Comment by davidjosephhayes** [comment]
How do I do know what to fix? When I click the Resolive Conflicts, nothing loads.

**Comment by wilsonge** [comment]
I should have fixed conflicts here. Can I have one tester to confirm this PR still works please

**Comment by RonakParmar** [comment]
I have tested this item :white_check_mark: successfully on fc626b1157b2b7ff587a964c3f8cc9e769f71fb1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12464">issues.joomla.org/tracker/joomla-cms/12464</a>.</sub>

**Pull request merged**